examples/quickstart-xgboost/client.py

```diff
@@ -0,0 +1,173 @@
+import warnings
+from logging import INFO
+import xgboost as xgb
+
+import flwr as fl
+from flwr_datasets import FederatedDataset
+from flwr.common.logger import log
+from flwr.common import (
+    Code,
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    GetParametersIns,
+    GetParametersRes,
+    Parameters,
+    Status,
+)
+
+from dataset import (
+    instantiate_partitioner,
+    train_test_split,
+    transform_dataset_to_dmatrix,
+    resplit,
+)
+from utils import client_args_parser
+
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+# Parse arguments for experimental settings
+args = client_args_parser()
+
+# Load (HIGGS) dataset and conduct partitioning
+num_partitions = args.num_partitions
+# Partitioner type is chosen from ["uniform", "linear", "square", "exponential"]
+partitioner_type = args.partitioner_type
+
+# Instantiate partitioner
+partitioner = instantiate_partitioner(
+    partitioner_type=partitioner_type, num_partitions=num_partitions
+)
+fds = FederatedDataset(
+    dataset="jxie/higgs", partitioners={"train": partitioner}, resplitter=resplit
+)
+
+# Let's use the first partition as an example
+partition_id = args.partition_id
+partition = fds.load_partition(idx=partition_id, split="train")
+partition.set_format("numpy")
+
+if args.centralised_eval:
```

Comment:
Just one more question. In the case of centralized eval each of the (federated) nodes also uses centralized dataset for the federated evaluation. Is that intended, or is it controlled in the server?